service/controller/v15/resource/pod/delete.go

```diff
@@ -51,6 +51,14 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 			resourcecanceledcontext.SetCanceled(ctx)
 			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
 
+			return nil
+		}
+		areContainersTerminated := key.ArePodContainersTerminated(currentPod)
+		if areContainersTerminated {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "all containers in pod are terminated; considering it drained")
```

Comment:
Having semicolon in log message doesn't look good.

Maybe:

```
r.logger.LogCtx(ctx, "level", "debug", "message", "pod is already drained because all containers are terminated")
```

Or adding a reason log as described here https://github.com/giantswarm/fmt/blob/master/go/log_messages.md#ensuring-state:
```
r.logger.LogCtx(ctx, "level", "debug", "message", "pod is already drained because")
r.logger.LogCtx(ctx, "level", "debug", "message", "all pod's containers are terminated")
```

Reply:
I would prefer the two logs but without `because`.

Reply:
I really dislike split logs because as a practice they are so wrong (in case there's every any concurrency either within service or with multiple services having log aggregation, it's impossible to make any sense out of interleaved messages) but I'll change it for now as we do this elsewhere as well.